firewood/src/db.rs

```diff
@@ -380,6 +383,34 @@ impl Drop for DbInner {
     }
 }
 
+#[async_trait]
+impl api::Db for Db {
+    type Historical = DbRev<SharedStore>;
+
+    type Proposal = Proposal;
+
+    async fn revision(&self, root_hash: HashKey) -> Result<Arc<Self::Historical>, api::Error> {
+        if let Some(rev) = self.get_revision(&TrieHash(root_hash)) {
+            Ok(Arc::new(rev.rev))
```

Comment:
I think this is correct for this PR, but it seems like we've got some more refactoring to do at some point. All these nested types drive me bananas.